plugins/module_utils/network/sonic/config/dcbx/dcbx.py

```diff
@@ -0,0 +1,440 @@
+    dcbx_intf_path = 'data/openconfig-dcbx:dcbx/interfaces/interface={intf_name}'
+
+    dcbx_intf_config_path = {
+        'enabled': dcbx_intf_path + '/config/enabled',
+        'pfc-tlv-enabled': dcbx_intf_path + '/config/pfc-tlv-enabled',
+        'ets-configuration-tlv-enabled': dcbx_intf_path + '/config/ets-configuration-tlv-enabled',
+        'ets-recommendation-tlv-enabled': dcbx_intf_path + '/config/ets-recommendation-tlv-enabled'
+    }
+
+    def __init__(self, module):
+        super(Dcbx, self).__init__(module)
+
+    def get_dcbx_facts(self):
+        """ Get the 'facts' (the current configuration)
+        :rtype: A dictionary
+        :returns: The current configuration as a dictionary
+        """
+        facts, _warnings = Facts(self._module).get_facts(
+            self.gather_subset, self.gather_network_resources)
+        dcbx_facts = facts['ansible_network_resources'].get('dcbx')
+        if not dcbx_facts:
+            return {}
+        return dcbx_facts
+
+    def execute_module(self):
+        """ Execute the module
+        :rtype: A dictionary
+        :returns: The result from module execution
+        """
+        result = {'changed': False}
+        warnings = []
+        commands = []
+
+        existing_dcbx_facts = self.get_dcbx_facts()
+        commands, requests = self.set_config(existing_dcbx_facts)
+        if commands and len(requests) > 0:
+            if not self._module.check_mode:
+                try:
+                    edit_config(self._module, to_request(
+                        self._module, requests))
+                except ConnectionError as exc:
+                    self._module.fail_json(msg=str(exc), code=exc.code)
+            result['changed'] = True
+
+        result['commands'] = commands
+        result['before'] = existing_dcbx_facts
+
+        new_config = {}
+        if self._module.check_mode:
+            state = self._module.params['state']
+            new_config = get_new_config(commands, existing_dcbx_facts,
+                                        TEST_KEYS_generate_config)
+
+            result['after(generated)'] = [new_config]
+        else:
+            new_config = self.get_dcbx_facts()
+            result['after'] = new_config
+
+        if self._module._diff:
+            result['diff'] = get_formatted_config_diff(existing_dcbx_facts,
+                                                       new_config,
+                                                       self._module._verbosity)
+
+        result['warnings'] = warnings
+        return result
+
+    def set_config(self, existing_dcbx_facts):
+        """ Collect the configuration from the args passed to the module,
+            collect the current configuration (as a dict from facts)
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        want = remove_empties(self._module.params['config'])
```

Comment:
For this argspec, this call isn't needed. If the Ansible "remove_empties" function (or the similar enterprise_sonic "remove_none") were called here, any deletion requests for deletion of all configuration on an interface by specifying the interface name with no config would get deleted. Instead, it should be fine to just do the following here:

```suggestion
        want = self._module.params['config']
```

Reply:
Done.